### PR TITLE
Adding OS Release and OS Build info to the run_info for Edge

### PR DESCRIPTION
**Pull request by stanleyhon**
Adding detailed OS Release and OS build info to the run_info for Edge runs.

### CODE REVIEW

**Comment by stanleyhon** [comment]
@foolip @mustjab @thejohnjansen

**Comment by foolip** [comment]
I'll leave this review to @gsnedders.

File: tools/wptrunner/wptrunner/browsers/edge.py [hunk 1]
```diff
@@ -6,6 +6,9 @@
                                           SeleniumRefTestExecutor)  # noqa: F401
 from ..executors.executoredge import EdgeDriverWdspecExecutor  # noqa: F401
 
+import subprocess
+from tools.wpt.utils import call
```
Comment (gsnedders) [inline tools/wptrunner/wptrunner/browsers/edge.py]: Let's not rely on `tools.wpt` here; just use `subprocess.check_output` without that wrapper. (I think there's still some use of wptrunner with it actually installed, and being able to import this relies on `tools` being on the path.)

File: tools/wptrunner/wptrunner/browsers/edge.py [hunk 2]
```diff
@@ -6,6 +6,9 @@
                                           SeleniumRefTestExecutor)  # noqa: F401
 from ..executors.executoredge import EdgeDriverWdspecExecutor  # noqa: F401
 
+import subprocess
```
Comment (gsnedders) [inline tools/wptrunner/wptrunner/browsers/edge.py]: Minor nit: standard Python practice would be to put this standard library import before all our local imports (so after the magic `from __future__ …` but before `from .base …`).

File: tools/wptrunner/wptrunner/browsers/edge.py [hunk 3]
```diff
@@ -97,3 +101,18 @@ def cleanup(self):
...
+        os_release = call("powershell.exe", osReleaseCommand).strip()
+        os_build = call("powershell.exe", osBuildCommand).strip()
+    except (subprocess.CalledProcessError, OSError):
+        return {}
+
+    rv = {
+        "os_build": os_build,
+        "os_release": os_release}
+
+    return rv
```
Comment (gsnedders) [inline tools/wptrunner/wptrunner/browsers/edge.py]: Make sure there's a trailing new line, plz.

File: tools/wptrunner/wptrunner/browsers/edge.py [hunk 4]
```diff
@@ -97,3 +101,18 @@ def cleanup(self):
...
+    osBuildCommand = "(Get-ItemProperty 'HKLM:\Software\Microsoft\Windows NT\CurrentVersion').BuildLabEx"
+    try:
+        os_release = call("powershell.exe", osReleaseCommand).strip()
+        os_build = call("powershell.exe", osBuildCommand).strip()
+    except (subprocess.CalledProcessError, OSError):
+        return {}
+
+    rv = {
+        "os_build": os_build,
+        "os_release": os_release}
```
Comment (gsnedders) [inline tools/wptrunner/wptrunner/browsers/edge.py]: With the `{` on a new line, I'd rather the `}` be on one too.

**Review by gsnedders — CHANGES_REQUESTED** [review]
This looks mostly good; one real comment (about `tools.wpt`) and a few minor nits. Thanks!

File: tools/wptrunner/wptrunner/browsers/edge.py (hunk 4 above)
Comment (stanleyhon) [reply tools/wptrunner/wptrunner/browsers/edge.py]: I've fixed this up to be formatted the same as the firefox file (open brace on same line, in-line key values, close brace on same line)

File: tools/wptrunner/wptrunner/browsers/edge.py (hunk 3 above)
Comment (stanleyhon) [reply tools/wptrunner/wptrunner/browsers/edge.py]: Fixed.

File: tools/wptrunner/wptrunner/browsers/edge.py (hunk 2 above)
Comment (stanleyhon) [reply tools/wptrunner/wptrunner/browsers/edge.py]: Moved.

File: tools/wptrunner/wptrunner/browsers/edge.py (hunk 1 above)
Comment (stanleyhon) [reply tools/wptrunner/wptrunner/browsers/edge.py]: Using subprocess.check_output.

**Comment by stanleyhon** [comment]
@gsnedders is there anything else I need to do? I've addressed your previous comments.

I noticed some automated failures, but they don't seem related to my changes.

**Comment by gsnedders** [comment]
@stanleyhon I think I got side-tracked into figuring out what was causing those failures, sorry!

**Review by gsnedders — APPROVED** [review]

File: tools/wptrunner/wptrunner/browsers/edge.py [hunk 5]
```diff
@@ -97,3 +99,17 @@ def cleanup(self):
 
     def executor_browser(self):
         return ExecutorBrowser, {"webdriver_url": self.server.url}
+
+
+def run_info_extras(**kwargs):
+    osReleaseCommand = "(Get-ItemProperty 'HKLM:\Software\Microsoft\Windows NT\CurrentVersion').ReleaseId"
```
Comment (Ms2ger) [inline tools/wptrunner/wptrunner/browsers/edge.py]:
```
  /home/test/web-platform-tests/tools/wptrunner/wptrunner/browsers/edge.py:105: DeprecationWarning: invalid escape sequence \S
```

I guess this should have used double backslashes.

Comment (gsnedders) [reply tools/wptrunner/wptrunner/browsers/edge.py]: Should probably be a raw string.

Comment (stanleyhon) [reply tools/wptrunner/wptrunner/browsers/edge.py]: Do I need to make a fresh PR to fix this up?

Comment (Ms2ger) [reply tools/wptrunner/wptrunner/browsers/edge.py]: If you could, yes, please.

Comment (stanleyhon) [reply tools/wptrunner/wptrunner/browsers/edge.py]: I've put up a CR over [here ](https://github.com/web-platform-tests/wpt/pull/16234)

**Pull request merged**